### ISSUE TITLE
Confusing error message

### ISSUE THREAD

**Issue by greg-hellings** [ansible/galaxy#2289]
##### SUMMARY
Spurious error reporting module name errors

##### STEPS TO REPRODUCE

* Create a collection
* Create a plugin module file
* Give the module name something like "__foo_module"
* Build and publish collection

##### EXPECTED RESULTS
Import succeeds

##### ACTUAL RESULTS
Import fails with the following message

```
ERROR! Galaxy import process failed: module has invalid name: '__jenkins_script', expecting name to contain only lowercase alphanumeric characters or '_' (Code: GLW0002)
```

Looking into galaxy/constants.py I see the following definition of valid identifiers

```python
CONTENT_NAME_REGEXP = re.compile(r'^(?!.*__)[a-z_]+[0-9a-z_]*$')
```

If my regex foo is correct then the restrictions are significantly more complex than just alpha-numerics and underscores. It's more like "only lowercase alphanumeric characters or '_', and cannot start with a number nor contain a double underscore anywhere."

**Comment by chouseknecht** [comment]
If you're attempting to deprecate a plugin, then I believe you use a single leading `_`, which is supported. Try using `_jenkins_script`, for example.

**Comment by greg-hellings** [comment]
Yes, in this case I didn't need the module at all. But the error message was still confusing (and, one might say, incorrect).
